### ISSUE TITLE
[Expo 50] Regression in `Asset selection`

### ISSUE THREAD

**Issue by o-alexandrov** [expo/expo#27065]
### Minimal reproducible example

https://github.com/o-alexandrov/expo-assets-regression

### Summary

Affected platforms `iOS`, `Android`.
Previously, afaik, it was possible to select extra assets with `assetBundlePatterns`.

Since [assetBundlePatterns is deprecated](https://github.com/expo/expo/pull/26962#issuecomment-1939453429), it's not possible to add assets that are NOT referenced with `require('path_to_file')`.
The new assetPatternsToBeBundled from dos on [Asset selection and exclusion](https://docs.expo.dev/eas-update/asset-selection/) offers only `exclusion` (not `selection`).

The provided minimal reproduction example shows an attempt to add `assets/example-extra-assets/example.txt` into the bundle, but `npx expo export --dump-assetmap` shows it's not selected.

Please add a way to copy files without a necessity to `require` them.
Some files MUST NOT be `require`d, as they would conflict with `metro`'s [assetExts](https://github.com/facebook/metro/blob/7028b7f51074f9ceef22258a8643d0f90de2388b/packages/metro-config/src/defaults/defaults.js#L15) and `metro` might fail to process such files. In such a case, the files must be just copied without extra processing and that's where I hope the new API `assetPatternsToBeBundled` comes to play (w/ optional property to pass through `metro` or not)

### Environment

<details>
  <summary>Click to expand</summary>
  
expo-env-info 1.2.0 environment info:
    System:
      OS: macOS 14.2.1
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 21.6.1 - /opt/homebrew/bin/node
      Yarn: 1.22.21 - /opt/homebrew/bin/yarn
      npm: 10.2.4 - /opt/homebrew/bin/npm
    Managers:
      CocoaPods: 1.15.2 - /opt/homebrew/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 23.2, iOS 17.2, macOS 14.2, tvOS 17.2, watchOS 10.2
    IDEs:
      Android Studio: 2022.3 AI-223.8836.35.2231.11090377
      Xcode: 15.1/15C65 - /usr/bin/xcodebuild
    npmPackages:
      expo: ~50.0.6 => 50.0.6 
      react: 18.2.0 => 18.2.0 
      react-native: 0.73.4 => 0.73.4 
    Expo Workflow: managed
</details>

**Comment by o-alexandrov** [comment]
Closing because there's Expo's provided example which doesn't seem to work
https://github.com/expo/UpdatesAPIDemo/issues/14

**Comment by douglowder** [comment]
@o-alexandrov after discussing this issue internally, we are not sure if the regression you are seeing is with assets included in the native build, or assets included in an EAS update.

Thanks for providing a sample project -- could you add a README there, describing the steps needed to show the problem, and describing what the old behavior was that has regressed?

If we have this information, we will be able to reopen this issue and investigate further.

**Comment by o-alexandrov** [comment]
@douglowder I realized I'm incapable of achieving what I need with `expo-updates`.
I need to include a single page application (SPA) website into an expo bundle and reference this application using a `file` protocol or, in other words, `uri` option of [react-native-webview](https://github.com/react-native-webview/react-native-webview).
- this procedure of  is described in [this article](https://yelotofu.com/react-native-load-local-static-site-inside-webview-2b93eb1c4225)

I need to add `assets` into a bundle without having to `require` them.
- in other words, I just want `expo` to copy files (without any processing) similar to [WebpackCopyPlugin](https://webpack.js.org/plugins/copy-webpack-plugin/) and other plugins in other bundlers

If I require `assets`, they go through `metro` bundler and `metro` breaks due to metro treating `.js` files as part of [assetExts](https://github.com/facebook/metro/blob/7028b7f51074f9ceef22258a8643d0f90de2388b/packages/metro-config/src/defaults/defaults.js#L15)

**Comment by douglowder** [comment]
@o-alexandrov agreed, `expo-updates` won't help you, especially since you need the assets in the original native build, not just over the air updates.

This seems like a new feature request, I will ask internally if there are any plans to do anything like this.

**Comment by marqroldan** [comment]
> @o-alexandrov agreed, `expo-updates` won't help you, especially since you need the assets in the original native build, not just over the air updates.
> 
> This seems like a new feature request, I will ask internally if there are any plans to do anything like this.

Sorry for replying to this issue after 3 months, but I'm interested in a solution as well

(Apologies if there's a latest one, I've been spread too thin tracking down a method to do such without falling back to a config plugin)
